runtime/jcl/common/sun_reflect_ConstantPool.c

```diff
@@ -919,6 +927,12 @@ registerJdkInternalReflectConstantPoolNatives(JNIEnv *env) {
 	/* jdk.internal.reflect.ConstantPool is currently cached in CLS_sun_reflect_ConstantPool */
 	jclass jdk_internal_reflect_ConstantPool = JCL_CACHE_GET(env, CLS_sun_reflect_ConstantPool);
 
+	/* This is temporary code, will be removed in the future */
+	if (J9_ARE_ANY_BITS_SET(vm->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_REFLECT_CONSTANTPOOL_REGISTER_NATIVES_CALLED)) {
+		goto _end;
+	}
+	vm->extendedRuntimeFlags2 |= J9_EXTENDED_RUNTIME2_REFLECT_CONSTANTPOOL_REGISTER_NATIVES_CALLED;
```

Comment:
What ensures this is thread-safe?

Reply:
JVM init is single threaded, so is class init.